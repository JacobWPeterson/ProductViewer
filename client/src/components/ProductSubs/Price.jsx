import React from 'react';
import styled from 'styled-components';

const NoSale = styled.p`
  margin: 0;
  margin-bottom: 1vh;
  font-weight: 300;
`;

const Discounted = styled.div`
  margin-bottom: 1vh;
  display: flex;
`;

const SaleReg = styled.p`
  margin: 0;
  font-weight: 300;
  text-decoration: line-through;
`;

const SalePrice = styled.p`
  margin: 0;
  padding-left: 1vw;
  font-weight: 300;
  color: red;
`;

const Price = ({ price, sale }) => {
  if (!sale) {
    return (
      <NoSale>{`$${Math.round(price)}`}</NoSale>
    );
  }
  return (
    <Discounted>
      <SaleReg>{`$${Math.round(price)}`}</SaleReg>
      <SalePrice>{`$${Math.round(sale)}`}</SalePrice>
    </Discounted>
  );
};

export default Price;
