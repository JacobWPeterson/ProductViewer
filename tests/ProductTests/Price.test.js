/* eslint-disable no-undef */
/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { mount } from 'enzyme';

import Price from '../../client/src/components/ProductSubs/Price.jsx';
import ProductTestData from '../ProductTestData.js';

describe('Price rendering', () => {
  const props = {
    price: ProductTestData.productStyles.results[0].original_price,
    sale: ProductTestData.productStyles.results[0].sale_price,
  };

  const PriceComponent = mount(<Price {...props} debug />);
  const text = (PriceComponent.find('p').text());

  it('Should render correctly', () => {
    expect(PriceComponent).toMatchSnapshot();
  });

  it('render the price correctly', () => {
    expect(text).toEqual('$40');
  });
});

describe('Price rendering', () => {
  const props = {
    price: ProductTestData.productStyles.results[2].original_price,
    sale: ProductTestData.productStyles.results[2].sale_price,
  };
  const PriceComponent = mount(<Price {...props} debug />);
  const RegPrice = (PriceComponent.find('p').at(0).text());
  const SalePrice = (PriceComponent.find('p').at(1).text());

  it('Should render correctly', () => {
    expect(PriceComponent).toMatchSnapshot();
  });

  it('render the regular price correctly', () => {
    expect(RegPrice).toEqual('$40');
  });

  it('render the sale price correctly', () => {
    expect(SalePrice).toEqual('$35');
  });
});
