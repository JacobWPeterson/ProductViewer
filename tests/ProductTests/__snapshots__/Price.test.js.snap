// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Price rendering Should render correctly 1`] = `
<Price
  debug={true}
  price="40.00"
  sale={null}
>
  <Price__NoSale>
    <p
      className="Price__NoSale-eho440-0 ituvCN"
    >
      $40
    </p>
  </Price__NoSale>
</Price>
`;

exports[`Price rendering Should render correctly 2`] = `
<Price
  debug={true}
  price="40.00"
  sale="35.00"
>
  <Price__Discounted>
    <div
      className="Price__Discounted-eho440-1 kJaIWp"
    >
      <Price__SaleReg>
        <p
          className="Price__SaleReg-eho440-2 fMsqin"
        >
          $40
        </p>
      </Price__SaleReg>
      <Price__SalePrice>
        <p
          className="Price__SalePrice-eho440-3 fwUqGd"
        >
          $35
        </p>
      </Price__SalePrice>
    </div>
  </Price__Discounted>
</Price>
`;
